uhttpd -m: allow the 2nd arg, if given, to be an existing hash to check the
password against, instead of always hashing with a fresh "$1$" salt.

--- a/main.c
+++ b/main.c
@@ -469,7 +469,7 @@ int main(int argc, char **argv)
 
 		/* md5 crypt */
 		case 'm':
-			printf("%s\n", crypt(optarg, "$1$"));
+			printf("%s\n", crypt(optarg, (optind < argc) ? argv[optind] : "$1$"));
 			return 0;
 			break;
 
